### ISSUE TITLE
How to run codeinterpreter-api ?

### ISSUE THREAD

**Issue by ElioErrico** [shroominic/codeinterpreter-api#162]
Hello everyone,

I am having touble on running the code.
This is my installing procedure.

1. `git clone https://github.com/shroominic/codeinterpreter-api.git`
2. let's go inside the folder
3. rename .env and set OPENAI_API_KEY
4. create a virtual enviroment
5. pip install codeinterpreterapi
6. go inside examples folder and try to run one

I Always get the error below.

Did I make some mistake or there is any bug?

```
LangChainDeprecationWarning: As of langchain-core 0.3.0, LangChain uses pydantic v2 internally. The langchain_core.pydantic_v1 module was a compatibility shim for pydantic v1, and should no longer be used. Please update the code to import from Pydantic directly.

For example, replace imports like: `from langchain_core.pydantic_v1 import BaseModel`
with: `from pydantic import BaseModel`
or the v1 compatibility namespace if you are working in a code base that has not been fully upgraded to pydantic 2 yet.         from pydantic.v1 import BaseModel

  from codeinterpreterapi.config import settings
Traceback (most recent call last):
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\examples\analyze_dataset.py", line 1, in <module>
    from codeinterpreterapi import CodeInterpreterSession, File
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeinterpreterapi\__init__.py", line 5, in <module>
    from codeinterpreterapi.session import CodeInterpreterSession
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeinterpreterapi\session.py", line 21, in <module>
    from langchain_community.chat_message_histories.in_memory import ChatMessageHistory
ModuleNotFoundError: No module named 'langchain_community'

```

Than if I install Langchain_comunity ....

```

LangChainDeprecationWarning: As of langchain-core 0.3.0, LangChain uses pydantic v2 internally. The langchain_core.pydantic_v1 module was a compatibility shim for pydantic v1, and should no longer be used. Please update the code to import from Pydantic directly.

For example, replace imports like: `from langchain_core.pydantic_v1 import BaseModel`
with: `from pydantic import BaseModel`
or the v1 compatibility namespace if you are working in a code base that has not been fully upgraded to pydantic 2 yet.         from pydantic.v1 import BaseModel

  from codeinterpreterapi.config import settings
Traceback (most recent call last):
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\show_bitcoin_chart.py", line 3, in <module>
    with CodeInterpreterSession() as session:
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeinterpreterapi\session.py", line 70, in __init__
    self.llm: BaseLanguageModel = llm or self._choose_llm()
                                         ^^^^^^^^^^^^^^^^^^
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeinterpreterapi\session.py", line 149, in _choose_llm
    return ChatOpenAI(
           ^^^^^^^^^^^
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\langchain_core\load\serializable.py", line 125, in __init__
    super().__init__(*args, **kwargs)
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\pydantic\main.py", line 212, in __init__
    validated_self = self.__pydantic_validator__.validate_python(data, self_instance=self)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
pydantic_core._pydantic_core.ValidationError: 1 validation error for ChatOpenAI
api_key
  Input should be a valid string [type=string_type, input_value=SecretStr('**********'), input_type=SecretStr]
    For further information visit https://errors.pydantic.dev/2.9/v/string_type
```

**Comment by shroominic** [comment]
Hey sorry for the inconvenience i this was caused by some dependency issues due to newer versions I think I fixed it now can you try the new version 0.1.19 and tell me if it works?

**Comment by ElioErrico** [comment]
Still have some problems!
Sorry if long message but I am trying to give as much information as I can.
Let me know !!!  I am very curious about your program!

I did install `codeinterpreterapi==0.1.19`

Now I get the following error:

```
py show_bitcoin_chart.py
C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeinterpreterapi\__init__.py:3: 

LangChainDeprecationWarning: As of langchain-core 0.3.0, LangChain uses pydantic v2 internally. The langchain_core.pydantic_v1 module was a compatibility shim for pydantic v1, and should no longer be used. Please update the code to import from Pydantic directly.

For example, replace imports like: `from langchain_core.pydantic_v1 import BaseModel`
with: `from pydantic import BaseModel`
or the v1 compatibility namespace if you are working in a code base that has not been fully upgraded to pydantic 2 yet.         from pydantic.v1 import BaseModel

  from codeinterpreterapi.config import settings
Make sure 'jupyter-kernel-gateway' is installed when using without a CODEBOX_API_KEY.
You can install it with 'pip install jupyter-kernel-gateway'.
Traceback (most recent call last):
  File "C:\Users\erric\AppData\Local\Programs\Python\Python311\Lib\importlib\metadata\__init__.py", line 563, in from_name
    return next(cls.discover(name=name))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\show_bitcoin_chart.py", line 3, in <module>
    with CodeInterpreterSession() as session:
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeinterpreterapi\session.py", line 488, in __enter__
    self.start()
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeinterpreterapi\session.py", line 89, in start
    status = SessionStatus.from_codebox_status(self.codebox.start())
                                               ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeboxapi\box\localbox.py", line 74, in start
    self._check_installed()
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeboxapi\box\localbox.py", line 158, in _check_installed
    distribution("jupyter-kernel-gateway")
  File "C:\Users\erric\AppData\Local\Programs\Python\Python311\Lib\importlib\metadata\__init__.py", line 981, in distribution
    return Distribution.from_name(distribution_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\erric\AppData\Local\Programs\Python\Python311\Lib\importlib\metadata\__init__.py", line 565, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: No package metadata was found for jupyter-kernel-gateway
```

Than I installed jupyter-kernel-gateway and I  run the code and I get following errors:

This in the shell:
```
[I 2024-11-03 11:58:32.781 KernelGatewayApp] Jupyter Kernel Gateway 3.0.1 is available at http://0.0.0.0:8888
[I 241103 11:58:33 web:2348] 200 GET /api (127.0.0.1) 1.04ms
[W 2024-11-03 11:58:35.207 KernelGatewayApp] Kernelspec name python3 cannot be found!
[E 2024-11-03 11:58:35.207 KernelGatewayApp] No such kernel named python3
    Traceback (most recent call last):
      File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\jupyter_client\manager.py", line 87, in wrapper
        out = await method(self, *args, **kwargs)
```
...

```
File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\jupyter_client\kernelspec.py", line 285, in get_kernel_spec
        raise NoSuchKernel(kernel_name)
    jupyter_client.kernelspec.NoSuchKernel: No such kernel named python3
[E 241103 11:58:35 web:2348] 500 POST /api/kernels (127.0.0.1) 66.23ms
```
This in the terminal:

```
py show_bitcoin_chart.py
C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeinterpreterapi\__init__.py:3: LangChainDeprecationWarning: As of langchain-core 0.3.0, LangChain uses pydantic v2 internally. The langchain_core.pydantic_v1 module was a compatibility shim for pydantic v1, and should no longer be used. Please update the code to import from Pydantic directly.

For example, replace imports like: `from langchain_core.pydantic_v1 import BaseModel`
with: `from pydantic import BaseModel`
or the v1 compatibility namespace if you are working in a code base that has not been fully upgraded to pydantic 2 yet.         from pydantic.v1 import BaseModel

  from codeinterpreterapi.config import settings
Could not connect to kernel. HTTPConnectionPool(host='localhost', port=8888): Read timed out. (read timeout=60)
Could not connect to kernel. HTTPConnectionPool(host='localhost', port=8888): Read timed out. (read timeout=60)
Could not connect to kernel. HTTPConnectionPool(host='localhost', port=8888): Read timed out. (read timeout=60)
Traceback (most recent call last):
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\show_bitcoin_chart.py", line 6, in <module>
    with CodeInterpreterSession() as session:
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeinterpreterapi\session.py", line 488, in __enter__
    self.start()
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeinterpreterapi\session.py", line 89, in start
    status = SessionStatus.from_codebox_status(self.codebox.start())
                                               ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeboxapi\box\localbox.py", line 107, in start
    self._connect()
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeboxapi\box\localbox.py", line 128, in _connect
    raise Exception("Could not start kernel after multiple attempts")
Exception: Could not start kernel after multiple attempts
Exception ignored in: <function LocalBox.__del__ at 0x0000015EB8886980>
Traceback (most recent call last):
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeboxapi\box\localbox.py", line 634, in __del__
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeboxapi\box\localbox.py", line 576, in stop
OSError: [WinError 6] Handle non valido

```

Than I :
1) pip install ipykernel
2) python -m ipykernel install --user
3) And I run again the code with the following error

```
py show_bitcoin_chart.py                                                                                                                                    
C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeinterpreterapi\__init__.py:3: LangChainDeprecationWarning: As of langchain-core 0.3.0, LangChain uses pydantic v2 internally. The langchain_core.pydantic_v1 module was a compatibility shim for pydantic v1, and should no longer be used. Please update the code to import from Pydantic directly.

For example, replace imports like: `from langchain_core.pydantic_v1 import BaseModel`
with: `from pydantic import BaseModel`
or the v1 compatibility namespace if you are working in a code base that has not been fully upgraded to pydantic 2 yet.         from pydantic.v1 import BaseModel

  from codeinterpreterapi.config import settings
C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeinterpreterapi\session.py:216: LangChainDeprecationWarning: Please see the migration guide at: https://python.langchain.com/docs/versions/migrating_memory/
  memory=ConversationBufferMemory(
C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeinterpreterapi\session.py:429: LangChainDeprecationWarning: The method `Chain.run` was deprecated in langchain 0.1.0 and will be removed in 1.0. Use :meth:`~invoke` instead.
  response = self.agent_executor.run(input=user_request.content)
Error in CodeInterpreterSession: TypeError  - BaseModel.validate() takes 2 positional arguments but 3 were given
Traceback (most recent call last):
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\show_bitcoin_chart.py", line 10, in <module>
    response.files[0].show_image() # Show the chart image
    ~~~~~~~~~~~~~~^^^
IndexError: list index out of range
Exception ignored in: <function LocalBox.__del__ at 0x000001B57FC99BC0>
Traceback (most recent call last):
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeboxapi\box\localbox.py", line 634, in __del__
  File "C:\Users\erric\Desktop\Python\04-Codeinterpreter_v2\codeinterpreterapi\venv\Lib\site-packages\codeboxapi\box\localbox.py", line 583, in stop
OSError: [WinError 87] Parametro non corretto
```

Probably still some problem with dependencies or codeboxapi.

**Comment by shroominic** [comment]
Hey sorry I thought I fixed but was using a different version in my venv...
I now released v0.1.20 and this is hopefully the fix!
Let me know if anything else is broken!
